mgmtsystem/virtualcenter.py

```diff
@@ -572,30 +655,31 @@ def clone_vm(self, source, destination, resourcepool=None, datastore=None, power
             folder = source_template.parent.parent.vmParent
         except AttributeError:
             folder = source_template.parent
-
         progress_callback("Picked folder `{}`".format(folder.name))
 
         task = source_template.CloneVM_Task(folder=folder, name=destination, spec=vm_clone_spec)
-
-        def _check():
+        start_time = time.time()
+        delta = 0
+        while task.info.state not in {'queued', 'running'} and delta < provision_timeout:
```

Comment:
@rsnyman why don't refresh-able objects have some kind of inplace refresh method?

as far as i can tell this makes quite a few things harder 

however it also doesn't prevent a wait_for usage, just have a local 1 element list as workaround and replace the first element